oci/private/image_index.bzl

```diff
@@ -40,6 +65,9 @@ def _expand_image_to_args(image, expander):
     return args
 
 def _oci_image_index_impl(ctx):
+    if len(ctx.attr.platforms) > 0 and len(ctx.attr.images) != len(ctx.attr.platforms):
+        fail("platforms can only be specified when there is exactly one image in the images attribute.")
```

Comment:
```suggestion
        fail("If set, 'platforms' needs to have the same number of elements as 'images'.")
```

Reply:
No that’s no right, number of images is  len(platforms) * len(images), for only one it has to be exactly len(platforms)

Reply:
Ok, that makes sense. What is surprising to me is that the conditional doesn't seem to cover the failure message that's being displayed. Did you mean the following conditional instead?

```
if len(ctx.attr.platforms) > 0 and len(ctx.attr.images) != 1:
```

Reply:
```
oci_image_index(
  platforms = ["platform1", "platform2"]
  images = ["image1"]
)
```

Length of `ctx.attr.images` will be 2 even though there is one image, ctx.attr.images contains a matrix of `images * platforms`

Reply:
So its correct.

Reply:
Thanks for taking the time to explain this very slowly to me, I get it now 🙏